### ISSUE TITLE
Can GAE support edge_weight?

### ISSUE THREAD

**Issue by mrbeann** [rusty1s/pytorch_geometric#612]
## ❓ Questions & Help

Hi, I found an example in https://github.com/rusty1s/pytorch_geometric/blob/master/examples/autoencoder.py
It seems now the encoder GCN can support edge_weight, but does any decoder can handle it?

**Comment by rusty1s** [comment]
Yes, the encoder can support edge weights, but you need to write your own decoder and loss function, because the inner product decoder and the binary cross entropy loss only work for unweighted graphs.

**Comment by mrbeann** [comment]
Yes, is there any examples of this? I think it we can use some regression loss with inner product decoder, am I right?

**Comment by rusty1s** [comment]
No, there is no example, but it should be quite trivial to implement by following the code of the autoencoder module. I have never seen an inner product decoder in combination with a regression loss, but it may be worth a try. An alternative would be to simply use an MLP based on source and target node latent features.

**Comment by mrbeann** [comment]
"An alternative would be to simply use an MLP based on source and target node latent features." what this means? the MLP input embeddings and output the weight?

**Comment by rusty1s** [comment]
Yes, something like:
```python
class MLPDecoder(torch.nn.Module):
    ....
    def forward(self, z, edge_index):
        row, col = edge_index
        return self.mlp(torch.cat([z[row], z[col]], dim=-1))
```
